### PR TITLE
Improve memory usage of extraction, move some benchmarks to passing

### PR DESCRIPTION
**Pull request by oflatt**
This PR also, unfortunately, has to disable the function inlining code which has a performance bug

### CODE REVIEW

**Review by yihozhang — APPROVED** [review]
Looks good!

File: dag_in_context/src/greedy_dag_extractor.rs [hunk 1]
```diff
@@ -354,30 +355,59 @@ fn calculate_cost_set(
 
     let mut shared_total = NotNan::new(0.).unwrap();
     let mut unshared_total = info.cm.get_op_cost(&node.op);
-    let mut costs: HashMap<ClassId, NotNan<f64>> = HashMap::default();
+    let mut costs: HashTrieMap<ClassId, NotNan<f64>> = Default::default();
```
Comment (yihozhang) [inline dag_in_context/src/greedy_dag_extractor.rs:358]: Do you have a sense of how persistent data structures improve performance? Functional data structures in general have a much larger constant factor.

Comment (oflatt) [reply dag_in_context/src/greedy_dag_extractor.rs:358]: We get more sharing in memory- we didn't get much of a speedup as far as I can tell, but memory usage is halved

Comment (oflatt) [reply dag_in_context/src/greedy_dag_extractor.rs:358]: Actually perhaps we did get a big speedup- looking at the numbers below

**Pull request merged**